examples/agents-researcher/app/actions.ts

```diff
@@ -0,0 +1,55 @@
+"use server";
+
+import { PollResult, StepRecord } from "./types";
+import { Redis } from "@upstash/redis";
+
+export async function pollOutputs(workflowRunId: string): Promise<PollResult> {
+  const redis = Redis.fromEnv();
+
+  const progress = (await redis.get(`${workflowRunId}:progress`)) as
+    | string
+    | null;
```

Comment:
instead of using these `as` statements everywhere which doesn't look good, we can use redis.get<string | null>. same for the other commands below